src/lkl/disk_mem.c

```diff
@@ -22,10 +22,8 @@ static struct enclave_disk_config* get_disk_config(int fd)
 
 static int fd_get_capacity(struct lkl_disk disk, unsigned long long* res)
 {
-    off_t off;
-
     struct enclave_disk_config* disk_config;
-    if (disk_config = get_disk_config(disk.fd))
+    if ( (disk_config = get_disk_config(disk.fd)) != NULL)
```

Comment:
You should need only one of these changes (double brackets or explicit compare).